src/Enzo/enzo_EnzoMethodPmDeposit.cpp

```diff
@@ -117,45 +118,41 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
     block->lower(&xm,&ym,&zm);
     block->upper(&xp,&yp,&zp);
     block->cell_width(&hx,&hy,&hz);
-
-    // declare particle position arrays
-
-    ParticleDescr * particle_descr = cello::particle_descr();
-    Grouping * particle_groups = particle_descr->groups();
-
-    int num_mass = particle_groups->size("has_mass");
-
-    // Accumulate particle density using CIC
-
+    
+    // Get cosmological scale factors, if cosmology is turned on
     enzo_float cosmo_a=1.0;
     enzo_float cosmo_dadt=0.0;
     EnzoPhysicsCosmology * cosmology = enzo::cosmology();
-
+    
     if (cosmology) {
-
-      double time = block->time();
-      double dt   = block->dt();
-      cosmology->compute_expansion_factor (&cosmo_a,&cosmo_dadt,time+alpha_*dt);
-
+      cosmology->compute_expansion_factor(&cosmo_a,&cosmo_dadt,
+					  block->time() + alpha_*block->dt());
     }
-    if (rank >= 1) hx *= cosmo_a;
-    if (rank >= 2) hy *= cosmo_a;
-    if (rank >= 3) hz *= cosmo_a;
+
+    double inv_vol = 1.0;
+    if (rank >= 1) inv_vol /= hx;
+    if (rank >= 2) inv_vol /= hy;
+    if (rank >= 3) inv_vol /= hz;
```

Comment:
If my previous comment (about multiplying the cell widths by the scale factor) is wrong, then I have a minor stylistic suggestion. 

It might make the code easier to follow if you moved this block of code up to line 121 (before you handle the cosmology stuff). There are 2 benefit to this:
1. The only usage of `hx`, `hy`, and `hz` would occur right after the values are retrieved.
2. The only usage of the cosmology information is used (i.e. the line `const double dt = alpha_ * block->dt() / cosmo_a;`) would now occur right after the cosmology information is retrieved.

Of course, if you don't think that this improves things, feel free to ignore this